src/averageAges.js

```diff
@@ -15,11 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const allMen = people.filter(
+    person => person.sex === 'm'
+  );
+  const menOfTheCentury = allMen.filter(
+    person => Math.ceil(person.died / 100) === century
+  );
+  const men = century === undefined ? allMen : menOfTheCentury;
+
+  return men.reduce(
+    (sum, person) => sum + (person.died - person.born),
+    0
+  ) / men.length;
```

Comment:
можно еще так
```suggestion
  return men.reduce((sum, person) => (
    sum + (person.died - person.born)
  ), 0) / men.length;
```
так даже лучше